bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/handler/ProcessInfoHandler.java

```diff
@@ -0,0 +1,200 @@
+
+        Thing thing = getThing();
+        if (ThingStatus.ONLINE != thing.getStatus()) {
+            logger.debug("Cannot handle command. Thing is not ONLINE.");
+            return;
+        }
+
+        if (!(command instanceof RefreshType)) {
+            logger.debug("Unsupported command {}. Supported commands: REFRESH.", command);
+            return;
+        }
+
+        try {
+            switch (channelUID.getId()) {
+                case CHANNEL_NAME: {
+                    String state = systeminfo.getProcessName(pid);
+                    updateState(channelUID, new StringType(state));
+                    break;
+                }
+                case CHANNEL_PROCESS_LOAD: {
+                    BigDecimal state = systeminfo.getProcessCpuUsage(pid);
+                    updateState(channelUID, new QuantityType<>(state, SmartHomeUnits.PERCENT));
+                    break;
+                }
+                case CHANNEL_PROCESS_RESIDENT_MEMORY: {
+                    BigDecimal state = systeminfo.getProcessResidentMemory(pid);
+                    updateState(channelUID, new QuantityType<>(state, SmartHomeUnits.BYTE));
+                    break;
+                }
+                case CHANNEL_PROCESS_VIRTUAL_MEMORY: {
+                    BigDecimal state = systeminfo.getProcessVirtualMemory(pid);
+                    updateState(channelUID, new QuantityType<>(state, SmartHomeUnits.BYTE));
+                    break;
+                }
+                case CHANNEL_PROCESS_USER: {
+                    String state = systeminfo.getProcessUser(pid);
+                    updateState(channelUID, new StringType(state));
+                    break;
+                }
+                case CHANNEL_PROCESS_PATH: {
+                    String state = systeminfo.getProcessPath(pid);
+                    updateState(channelUID, new StringType(state));
+                    break;
+                }
+                case CHANNEL_PROCESS_THREADS: {
+                    BigDecimal state = systeminfo.getProcessThreads(pid);
+                    updateState(channelUID, new DecimalType(state));
+                    break;
+                }
+                default: {
+                    logger.debug("Channel with unknown ID: {}.", channelUID);
+                    break;
+                }
+            }
+        } catch (IllegalArgumentException exception) {
+            if (isLinked(channelUID)) {
+                logger.warn("No information for channel {} with process id {}.", channelUID, pid);
+            }
+            updateState(channelUID, UnDefType.UNDEF);
+        } catch (Exception exception) {
+            String message = exception.getMessage();
+            logger.debug("Unexpected error occurred while getting system information: {}.", message);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+        }
```

Comment:
If you don't expect any other exception here, you could simply let the exception uncought. The framework will log it. Also, it seems a bit unreasonable to set the status to offline when one info fails. This could lead to a flapping state if the other infos succeed.